### DISCUSSION TITLE
ERROR in FATES Plant Hydraulic

### DISCUSSION THREAD

**Discussion by LianGong98** [NGEET/fates#913] [Q&A]
Hi, when I run FTEAS with FATES_HYDRO in CESM2.2.0, using the default hydraulic parameters, I got the following error in lnd.log :
_EDPlantHydraulics water balance error exceeds threshold of = 
 1.000000000000000E-004 
 ENDRUN:
 ERROR in FatesPlantHydraulicsMod.F90 at line 3553_
I’ve turned up the value to 1.99 in _FatesPlantHydraulicsMod.F90_, but it still gives me this error, I do not know why it happened, and I would like to get your help, thank you very much!

**Comment by rgknox** [comment]
@LianGong98 there should be a few more lines of diagnostics that print out following this message, such as leaf_water: and the lat/lon.  Could you pass us that information too?  This error message is during the individual plant solve.  Depending on what we learn from the diagnostics, there are a few things that can be done.  Also, what is FTEAS and is this happening immediately in the simulation or does it seem to occur at a random time?
Thanks for posting

**Reply by LianGong98** [reply]
Hi, thanks for your reply. It seemed to occur at a random time? For example,
_FATES Dynamics: 2003-07-13
clm: leaving fates model           1
clm: completed timestep         9265
clm: completed timestep         9266
clm: completed timestep         9267
clm: completed timestep         9268
EDPlantHydraulics water balance error exceeds threshold of =1.000000000000000E-004
ENDRUN:
ERROR in FatesPlantHydraulicsMod.F90 at line 3553_
and I did not find other diagnostic information in this _lon.log file_, it is already at the end of this flie.

**Comment by rgknox** [comment]
Its tough to be absolutely sure what is happening without lots of diagnostics, but ...  Sometimes the model enounters a scenario where the transpiration generated over a time-step during the land-surface-atmosphere coupling is more than is available inside the leaf.  In this part of the code, the plant is trying to move water inside the plant in response to that transpiration flux, and simply can't accomodate the large transpiration demand.  

In the following PR, @xuchongang proposes a work-around:
https://github.com/NGEET/fates/pull/851/files#diff-e5e2459b995e00503077bce86074962e2acc4780ca00f248242b153527655c42R2597-R2604

Potential fixes: 
1) might want to try integrating the fix from @xuchongang (I think the idea is a good one, as a group we had concerns that the new error term might build up over time, but no evidence to suggest that was true)
2) try an alternative plant hydraulics solver, specifically, in the fates parameter file, try setting fates_hydro_solver = 2

https://github.com/NGEET/fates/blob/sci.1.59.6_api.24.1.0/parameter_files/fates_params_default.cdl#L669-L671

3) There are also tweaks that can be made to the plant conductance parameters or the parameters that handle the solver iterations and such, but it might be better to try options 1 and 2.  

Alternatively, @JunyanDing @pnlfang @xuchongang @mariuslam and others might have ideas?

**Reply by LianGong98** [reply]
Thank you very much! I'll try options 1 and 2.
